cpp/models/abm/infection.h

```diff
@@ -0,0 +1,142 @@
+/*
+* Copyright (C) 2020-2021 German Aerospace Center (DLR-SC)
+*
+* Authors: David Kerkmann, Sascha Korf
+*
+* Contact: Martin J. Kuehn <Martin.Kuehn@DLR.de>
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+#ifndef EPI_ABM_INFECTION_H
+#define EPI_ABM_INFECTION_H
+
+#include "abm/time.h"
+#include <vector>
+#include "abm/state.h"
+#include <memory>
+
+namespace mio
+{
+namespace abm
+{
+
+class Virus
+{
+public:
+    Virus(double infectivity, double mortality, double recovery)
+        : m_infectivity(infectivity)
+        , m_mortality(mortality)
+        , m_recovery(recovery)
+    {
+    }
+
+    double get_infectivity() const
+    {
+        return m_infectivity;
+    }
+
+    double get_mortality() const
+    {
+        return m_mortality;
+    }
+
+    double get_recovery() const
+    {
+        return m_recovery;
+    }
+
+private:
+    double m_infectivity;
+    double m_mortality;
+    double m_recovery;
+};
+
+class ViralLoad
+{
+public:
+    ViralLoad()
+    {
+        draw_viral_load();
```

Comment:
Doesnt this cause an error? Dont we need a forward declaration here?

I would also say that this can be done in the constructor itself without introducing an additional function.

Reply:
I put the implementation in the cpp file, shouldn't be an issue now.

I would postpone the discussion about needing an additional function to later, when we know how big the drawing method becomes. In general, I wouldn't mind to have a method for that to increase readability of the code.